### PR TITLE
bpo-42006: Stop using PyDict_GetItemString()

### PR DESCRIPTION
**Pull request by hongweipeng**
This is a complement to that PR #22648.

https://bugs.python.org/issue42006

### CODE REVIEW

File: Python/initconfig.c [hunk 1]
```diff
@@ -1050,7 +1050,7 @@ _PyConfig_AsDict(const PyConfig *config)
 static PyObject*
 config_dict_get(PyObject *dict, const char *name)
 {
-    PyObject *item = PyDict_GetItemString(dict, name);
+    PyObject *item = PyMapping_GetItemString(dict, name);
```
Comment (serhiy-storchaka) [inline Python/initconfig.c]: PyDict_GetItemString returns a borrowed reference, PyMapping_GetItemString returns an owned reference.

Comment (hongweipeng) [reply Python/initconfig.c]: Thanks for reminding, it has been replaced by `PyMapping_GetItemString`.

**Comment by serhiy-storchaka** [comment]
There are other issues in this code. See #23488.

**Comment by hongweipeng** [comment]
Superseded by #23488.

**Pull request closed without merging**